lib/main.js

```diff
@@ -319,6 +319,69 @@ if (options.interactiveTests) {
+      ],
+      do: [
+        'Copy and paste the below script into the shell to run it - it will take the device\'s credentials, calls the ' +
+        'API for the supervisor for cross-check, update the API entry for this device, and calls the supervisor updater ' +
+        'to check if it finds the updated info:\n\n' +
+        'source /etc/resin-supervisor/supervisor.conf\n' +
+        'if [ -f "/mnt/boot/config.json" ]; then\n' +
+        '  CONFIGJSON=/mnt/boot/config.json\n' +
+        'elif [ -f "/mnt/conf/config.json" ]; then\n' +
+        '  CONFIGJSON=/mnt/conf/config.json\n' +
+        'fi\n' +
+        'if [ -z "$SUPERVISOR_TAG" ]; then\n' +
+        '  echo "Couldn\'t extract SUPERVISOR_TAG from the supervisor.conf"\n' +
+        'elif [ -z "$CONFIGJSON" ]; then\n' +
+        '  echo "Couldn\'t find config.json, cannot continue".\n' +
+        'else\n' +
+        '  APIKEY="$(jq -r \'.apiKey // .deviceApiKey\' "${CONFIGJSON}")"\n' + // eslint-disable-line no-template-curly-in-string
+        '  DEVICEID="$(jq -r \'.deviceId\' "${CONFIGJSON}")"\n' + // eslint-disable-line no-template-curly-in-string
+        '  API_ENDPOINT="$(jq -r \'.apiEndpoint\' "${CONFIGJSON}")"\n' + // eslint-disable-line no-template-curly-in-string
+        '  SLUG="$(jq -r \'.deviceType\' "${CONFIGJSON}")"\n' + // eslint-disable-line no-template-curly-in-string
+        '  echo "SUPERVISOR_TAG: $SUPERVISOR_TAG"\n' +
+        // eslint-disable-next-line no-template-curly-in-string
+        '  SUPERVISOR_ID=$(curl -s "${API_ENDPOINT}/v3/supervisor_release?\\$select=id,image_name&\\$filter=((device_type%20eq' +
+        // eslint-disable-next-line no-template-curly-in-string
+        '%20\'$SLUG\')%20and%20(supervisor_version%20eq%20\'$SUPERVISOR_TAG\'))&apikey=${APIKEY}" | jq -e -r \'.d[0].id\')\n' +
+        '  echo "Extracted supervisor ID: $SUPERVISOR_ID"\n' +
+        // eslint-disable-next-line no-template-curly-in-string
+        '  API_REPLY=$(curl -s "${API_ENDPOINT}/v2/device($DEVICEID)?apikey=$APIKEY" -X PATCH -H \'Content-Type: application/' +
+        'json;charset=UTF-8\' --data-binary "{\\"supervisor_release\\": \\"$SUPERVISOR_ID\\"}")\n' +
+        '  if [ "$API_REPLY" = "OK" ]; then\n' +
+        '    if update-resin-supervisor | grep "Supervisor configuration found from API." ; then\n' +
+        '      TEST_OK=yes\n' +
+        '    fi\n' +
+        '  else\n' +
+        '    echo "Couldn\'t set Supervisor release in the API"\n' +
+        '  fi\n' +
+        'fi\n' +
+        'if [ "$TEST_OK" = "yes" ]; then\n' +
+        '  echo "Test SUCCESSFUL."\n' +
+        'else\n' +
+        '  echo "Test FAILED."\n' +
+        'fi'
+      ],
+      assert: [
+        'Setting the supervisor should output the supervisor release ID and an OK for the set: ' +
+        '"Extracted supervisor ID: 968"',
+        'If you see "Extracted supervisor ID: null", then check https://api.resin.io/v4/supervisor_release whether ' +
```

Comment:
I don't understand what do in this step. Lets say I get the `null` message. I then check the provided URL, and then what? Does it mean that I must mark the test as successful? 

Otherwise, if I always should mark the test as a failure if I get `null`, then why should I go checking `supervisor_release` if it will not make any difference in the test case result?

I think we should just tell the user that if they get `null` instead of a number, then that's a failure, without providing the URL and all the explanation.